### ISSUE TITLE
The rule `organizeDeclarations` adds redundant line if the original file has blank lines with spaces

### ISSUE THREAD

**Issue by igor-makarov** [nicklockwood/SwiftFormat#1809]
Sample source file (note the blank lines with whitespace):

```swift
public class TestClass {
    var variable01 = 1
    var variable02 = 2
    var variable03 = 3
    var variable04 = 4
    var variable05 = 5
    
    public func foo() {
    }
    
    internal func bar() {
    }
    
    private func baz() {
    }
}
```

Command to run:

```
swiftformat Testfile.swift --classthreshold 1 \
  --rules organizeDeclarations
```

Resulting file (note the redundant double lines):

```swift
public class TestClass {

    // MARK: Public

    
    public func foo() {
    }

    // MARK: Internal

    var variable01 = 1
    var variable02 = 2
    var variable03 = 3
    var variable04 = 4
    var variable05 = 5

    
    internal func bar() {
    }

    // MARK: Private

    
    private func baz() {
    }
}
```

**Comment by calda** [comment]
Thanks for the report! Here's a fix: https://github.com/nicklockwood/SwiftFormat/pull/1810

**Comment by igor-makarov** [comment]
Thanks @calda! 
I've spent half a day debugging this! 

We're adopting the ABNB SG and I've made two (temporary) rule changes to facilitate adoption. The problematic change was disabling the blank line trimming.

Turns out, SwiftFormat and SwiftLint would not interact well on this: 
* SF would add a redundant line 
* SL would remove a redundant line
* I would run the SG with `--lint` and it would complain over `organizeDeclarations` being wrong
 
It took me a while to realize that the line trimming would leave in the original code line with the whitespace (which SF didn't recognize as part of the mark comment).
